File: app/src/main/java/com/example/intern_anrdoid_2020/view/SearchFragment.java
package com.example.intern_anrdoid_2020.view;


import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import com.example.intern_anrdoid_2020.R;
import com.example.intern_anrdoid_2020.model.response.QiitaArticleResponse;

import java.util.ArrayList;

public class SearchFragment extends Fragment {

    private static final int PAGE = 1;
    private static final int PER_PAGE = 10;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActionBar actionBar = ((AppCompatActivity) getActivity()).getSupportActionBar();
        actionBar.setTitle(R.string.search_view);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        View v = inflater.inflate(R.layout.fragment_search, container, false);
        return setup(v);
    }

    private View setup(View v) {

        Button searchButton = (Button) v.findViewById(R.id.search_button);
        searchButton.setOnClickListener((View view) -> {
            String searchkay = getView().<EditText>findViewById(R.id.edit_search).getText().toString();
            QiitaListRepository.listArticle(PAGE, PER_PAGE, searchkay).observe(getViewLifecycleOwner(), (qiitaListResponse) -> {
                if (qiitaListResponse != null) {
                    showQiitaListFragment(qiitaListResponse);
                }
            });
        });
        return v;
    }

    private void showQiitaListFragment(ArrayList<QiitaArticleResponse> qiitaArticleResponse) {
        FragmentTransaction fragmentTransaction = getFragmentManager().beginTransaction();
        fragmentTransaction.replace(R.id.search_layout_frame, QiitaListFragment.newInstance(qiitaArticleResponse))
                           .addToBackStack(null)
                           .commit();
    }
}

File: app/src/main/java/com/example/intern_anrdoid_2020/view/SearchFragment.kt
package com.example.intern_anrdoid_2020.view

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import com.example.intern_anrdoid_2020.R
import com.example.intern_anrdoid_2020.model.response.QiitaArticleResponse
import java.util.*

class SearchFragment : Fragment() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val appCompatActivity = activity as AppCompatActivity?
        val actionBar = appCompatActivity?.supportActionBar
        actionBar?.setTitle(R.string.search_view)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val v = inflater.inflate(R.layout.fragment_search, container, false)
        return setup(v)
    }

    private fun setup(v: View): View {
        val searchButton = v.findViewById<View>(R.id.search_button) as Button
        searchButton.setOnClickListener {
            val searchkay = it.findViewById<EditText>(R.id.edit_search).text.toString()
            QiitaListRepository.listArticle(PAGE, PER_PAGE, searchkay).observe(viewLifecycleOwner, Observer { qiitaListResponse: ArrayList<QiitaArticleResponse> ->
                showQiitaListFragment(qiitaListResponse)
            })
        }
        return v
    }

    private fun showQiitaListFragment(qiitaArticleResponse: ArrayList<QiitaArticleResponse>) {
        fragmentManager?.let {
            val fragmentTransaction = it.beginTransaction()
            fragmentTransaction.replace(R.id.search_layout_frame, QiitaListFragment.newInstance(qiitaArticleResponse))
                    .addToBackStack(null)
                    .commit()
        }
    }

    companion object {
        private const val PAGE = 1
        private const val PER_PAGE = 10
    }
}
